### PR TITLE
Enable <Nullable> in .NET MAUI project templates

### LINKED ISSUE #17987: Enable <Nullable> in .NET MAUI project templates
Set `<Nullable>enable</Nullable>` in the .NET MAUI project templates (native, Blazor Hybrid, and library) to align it with C# best practices.

How to make things work in the MAUI templates: 

* The MAUI App template is already nullable-compatible, it's just not enabled
* The MAUI Blazor App template is _not_ nullable-compatible, but should be trivial to enable, because the ASP.NET Blazor Web App template it's based off of _is_ nullable-enabled
* The MAUI Library template is already nullable-compatible, it's just not enabled

As far as what the other cool kids around town are doing, every single project type I looked at is already nullable-enabled:

* Blazor Web App: Nullable-enabled
* ASP.NET Core Web App: Nullable-enabled
* Console app: Nullable-enabled
* WinForms app: Nullable-enabled (and arguably the _coolest_ kid in town)
* WPF app: Nullable-enabled
* MSTest Test Project: Nullable-enabled

So, it's probably worth doing in .NET MAUI unless someone can think of a good reason not to. For example, would the most value come only after we do a full nullable-annotation pass of .NET MAUI itself? Or is that not necessary?

### PR DESCRIPTION
**Pull request by Eilon**
Tested locally, everything builds and runs fine, no warnings. I also tried adding various MAUI *Item* templates to the nullable-enabled projects and they're all fine (they don't have much code in them to begin with).

Fixes #17987

### CODE REVIEW

File: src/Templates/src/templates/maui-blazor/MauiApp.1.csproj [hunk 1]
```diff
@@ -6,19 +6,20 @@
         <!-- Uncomment to also build the tizen app. You will need to install tizen by following this: https://github.com/Samsung/Tizen.NET -->
         <!-- <TargetFrameworks>$(TargetFrameworks);DOTNET_TFM-tizen</TargetFrameworks> -->
 
-         <!-- Note for MacCatalyst:
+        <!-- Note for MacCatalyst:
```
Comment (Eilon) [inline src/Templates/src/templates/maui-blazor/MauiApp.1.csproj:9]: Also fixed this spacing that was somehow off.

**Comment by Eilon** [comment]
Some seemingly legit test failures so far on Android:

```
   Restored /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj (in 2.24 sec).
 /Users/builder/azdo/_work/2/s/bin/dotnet/sdk/8.0.100-rtm.23506.1/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(311,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
 /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/Platforms/Android/Instrumentation.cs(17,30): error CS8765: Nullability of type of parameter 'arguments' doesn't match overridden member (possibly because of nullability attributes). [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
 /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/Platforms/Android/Instrumentation.cs(29,39): error CS8600: Converting null literal or possible null value to non-nullable type. [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
 /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/Platforms/Android/Instrumentation.cs(32,33): error CS8604: Possible null reference argument for parameter 'packageContext' in 'Intent Intent.SetClassName(Context packageContext, string className)'. [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
 /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/Platforms/Android/Instrumentation.cs(34,40): error CS8600: Converting null literal or possible null value to non-nullable type. [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
 Build FAILED.
 /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/Platforms/Android/Instrumentation.cs(17,30): error CS8765: Nullability of type of parameter 'arguments' doesn't match overridden member (possibly because of nullability attributes). [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
 /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/Platforms/Android/Instrumentation.cs(29,39): error CS8600: Converting null literal or possible null value to non-nullable type. [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
 /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/Platforms/Android/Instrumentation.cs(32,33): error CS8604: Possible null reference argument for parameter 'packageContext' in 'Intent Intent.SetClassName(Context packageContext, string className)'. [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
 /Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/Platforms/Android/Instrumentation.cs(34,40): error CS8600: Converting null literal or possible null value to non-nullable type. [/Users/builder/azdo/_work/_temp/test-intg/RunOnAndroidmauiblazornet80Release/RunOnAndroidmauiblazornet80Release.csproj::TargetFramework=net8.0-android]
     0 Warning(s)
     4 Error(s)
```

**Comment by mattleibow** [comment]
Ah, that may be the source generators for the tests... Should just be maui code, so we could slap a #nullable disable in there to be safe, or fix it and slap an enable...

**Comment by Eilon** [comment]
Yup doing that now

**Review by mattleibow — APPROVED** [review]

**Review by SteveSandersonMS — APPROVED** [review]

**Pull request merged**